packages/cactus-api-client/src/main/typescript/socketio-api-client.ts

```diff
@@ -0,0 +1,345 @@
+/*
+ * Copyright 2020-2021 Hyperledger Cactus Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * SocketIOApiClient.ts
+ */
+
+// TODO - Document it
+
+const defaultMaxCounterRequestID = 100;
+const defaultSyncFunctionTimeoutMillisecond = 5000;
+
+import { Logger, Checks } from "@hyperledger/cactus-common";
+import { LogLevelDesc, LoggerProvider } from "@hyperledger/cactus-common";
+import { ISocketApiClient } from "@hyperledger/cactus-core-api";
+
+import { Socket, SocketOptions, ManagerOptions, io } from "socket.io-client";
+import { readFileSync } from "fs";
+import { resolve as resolvePath } from "path";
+import { verify, VerifyOptions } from "jsonwebtoken";
+import { Observable, ReplaySubject } from "rxjs";
+import { finalize } from "rxjs/operators";
+
+// TODO - Remove any from here
+export function verifyValidatorJwt(
+  keyPath: string,
+  targetData: string,
+): Promise<any> {
+  return new Promise((resolve, reject) => {
+    const publicKey = readFileSync(resolvePath(__dirname, keyPath));
+
+    const option: VerifyOptions = {
+      algorithms: ["ES256"],
+    };
+
+    verify(targetData, publicKey, option, function (err: any, decoded: any) {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(decoded);
+      }
+    });
+  });
+}
+
+export type SocketIOApiClientOptions = {
+  readonly validatorID: string;
+  readonly validatorURL: string;
+  readonly validatorKeyPath: string;
+  readonly logLevel?: LogLevelDesc;
+  readonly maxCounterRequestID?: number;
+  readonly syncFunctionTimeoutMillisecond?: number;
+  readonly socketOptions?: Partial<ManagerOptions & SocketOptions>;
+};
+
+export class SocketLedgerEvent {
+  id = "";
+  verifierId = "";
+  data: Record<string, unknown> | null = null;
+}
+
+// TODO - To entire class - analyze and handle broken connection / socket disconnected scenarios
+export class SocketIOApiClient implements ISocketApiClient<SocketLedgerEvent> {
```

Comment:
@outSH Bringing this over from the other thread, how about `LegacyApiClient` for the name?

Reply:
I don't want jump on to your idea to change the name to LegacyApiClient. We may have some discussion to decide what name is the best one. But for this PR, can we focus on what is described in the commit message?

Reply:
@izuru0 Sure, I am.